dali/test/python/test_pipeline_multichannel.py

```diff
@@ -0,0 +1,247 @@
+
+class MultichannelPipeline(Pipeline):
+    def __init__(self, device, batch_size, num_threads=1, device_id=0):
+        super(MultichannelPipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+
+        self.reader = ops.FileReader(file_root=multichannel_tiff_root)
+
+        decoder_device = 'mixed' if self.device == 'gpu' else 'cpu'
+        self.decoder = ops.ImageDecoder(device = decoder_device, output_type = types.ANY_DATA)
+
+        self.resize = ops.Resize(device = self.device,
+                                 resize_y = 900, resize_x = 300,
+                                 min_filter=types.DALIInterpType.INTERP_LINEAR)
+
+        self.crop = ops.Crop(device = self.device,
+                             crop_h = 220, crop_w = 224,
+                             crop_pos_x = 0.3, crop_pos_y = 0.2,
+                             image_type = types.ANY_DATA)
+
+        self.transpose = ops.Transpose(device = self.device,
+                                       perm = (1, 0, 2),
+                                       transpose_layout = False)
+
+        self.cmn = ops.CropMirrorNormalize(device = self.device,
+                                           std = 255., mean = 0.,
+                                           output_layout = types.NHWC,
+                                           output_dtype = types.FLOAT)
+
+    def define_graph(self):
+        encoded_data, _ = self.reader()
+        decoded_data = self.decoder(encoded_data)
+        out = decoded_data.gpu() if self.device == 'gpu' else decoded_data
+        out = self.resize(out)
+        out = self.crop(out)
+        out = self.transpose(out)
+        out = self.cmn(out)
+        return out
+
+class MultichannelPythonOpPipeline(Pipeline):
+    def __init__(self, function, batch_size, num_threads=1, device_id=0):
+        super(MultichannelPythonOpPipeline, self).__init__(batch_size,
+                                                           num_threads,
+                                                           device_id,
+                                                           exec_async=False,
+                                                           exec_pipelined=False)
+        self.reader = ops.FileReader(file_root=multichannel_tiff_root)
+        self.decoder = ops.ImageDecoder(device = 'cpu', output_type = types.ANY_DATA)
+        self.oper = ops.PythonFunction(function=function)
+
+    def define_graph(self):
+        encoded_data, _ = self.reader()
+        decoded_data = self.decoder(encoded_data)
+        out = self.oper(decoded_data)
+        return out
+
+
+def check_full_pipe_multichannel_vs_numpy(device, batch_size):
+    compare_pipelines(MultichannelPipeline(device, batch_size),
+                      MultichannelPythonOpPipeline(full_pipe_func, batch_size),
+                      batch_size=batch_size, N_iterations=10,
+                      eps = 1e-03)
+
+def test_full_pipe_multichannel_vs_numpy():
```

Comment:
Do we need that anymore as L173 already covers that case?

Reply:
It doesn't. This tests uses ImageDecoder (not synthetic data)